test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Server/ActionOverloadingTestsController.cs

```diff
@@ -0,0 +1,303 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ActionOverloadingTestsController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server
+{
+    public class ActionOverloadingTestsController : ODataController

```

Comment:
Not for the first time, I have noticed `ActionOverloadingQueryTestsController` and `ActionOverloadingTestsController`. What is the difference between the two?

Reply:
They are not different per say but we have two test classes. `ActionOverloadingEndToEndTests` and `ActionOverloadingQueryTests` that test different things but since they share the same data source and they can run in parallel, they lead to data corruption. That's why have created a different controller for each test class although some of the APIs could be the same.